### ISSUE TITLE
Custom fonts

### ISSUE THREAD

**Issue by LeSniffa** [neanes/neanes#17]
**Is your feature request related to a problem? Please describe.**
Are there any plans to allow users to use custom fonts for mode keys, neumes etc? I have some installed fonts that I would like to use.
eg:
![image](https://user-images.githubusercontent.com/110656129/183015831-a2c95e1a-8922-4ede-af9b-fffb4a01dc0d.png)

**Comment by danielgarthur** [comment]
There are plans, but no other fonts are currently supported. The current font uses a font mapping called [SBMuFL](https://danielgarthur.github.io/sbmufl), which is inspired by [SMuFL](https://w3c.github.io/smufl/latest/index.html). The font was created using the glyphs from the EZ Psaltica font, re-mapped and modified to take advantage of OpenType features.

The main idea is that by using an OpenType font, characters such as the klasma and gorgon can be correctly positioned over each of the different base characters by the font itself, rather than relying on the a program like Neanes to position the neumes in the correct place.

Right now, there are no other SBMuFL fonts that I'm aware of. At some point, I would like to adapt a second font to use the SBMuFL mapping and document the process.

Since I don't have a timeline for that, is there anything that I could improve about the current font in the meantime? The current font seems to be very similar to the BZ fonts that you referenced above. Is it that you want a font with a different visual style? Or is it that there are characters that are missing from the current font that you want to use (e.g. oxeia)?

**Comment by LeSniffa** [comment]
The question was mostly motivated from the desire to use fonts that are more stylized to mimic the look of hand-written manuscripts. The font I provided as an example was indeed too similar and so I didn't really make my question clear. Sorry about that. 

When it comes to characters that might be missing, I have noticed a few that I would like to be added in the future if possible.
First is this neumen.

![neumen](https://user-images.githubusercontent.com/110656129/183889656-ba35c5b4-635d-43ea-b3af-eb4aa604bcc1.JPG)

I apologize but this stylized picture is the only example I have of it.
I would also like it if the corona and breath mark were added.

There also appear to be cases of characters overlapping each other like it happens with the gorgon in the example bellow.

![gorgon](https://user-images.githubusercontent.com/110656129/183890394-09d83888-70b5-4485-9b70-c082731bd9ec.JPG)

**Comment by danielgarthur** [comment]
The next release will have a "space after" property in the bottom toolbar that will let you add extra space after a neume to address the issue of the adjacent neumes clashing.
<img width="319" alt="image" src="https://user-images.githubusercontent.com/4132779/185757163-5732b936-0fc9-4be4-9621-a3b190b6393f.png">

**Comment by TugleaCristian01** [comment]
As a suggestion, I think the best neumes font to add to Neanes is "Chourmouzes", as it's among the most appreciated and nice fonts. How exactly does SBMuFL mapping of a font work? Can we help adapt other fonts to this format, so that they can be added to Neanes?

Thank you!

[Chourmouzez TTF+OFF.zip](https://github.com/danielgarthur/neanes/files/11354127/Chourmouzez.TTF%2BOFF.zip)

**Comment by basil** [comment]
Thank you! I used them [here](http://byzantinemusic.org/tmp/Axion%20Estin%20Pl%201%20Anton%20Pann.pdf).

**Comment by danielgarthur** [comment]
@neobabis I've finally gotten around to trying out this font in Neanes. It looks good in my tests so far. 

I think that it would be best to bundle the font with Neanes itself. Otherwise, let's say you create a score on your computer with your custom font, and you move around some fthores or other characters to make the score prettier. But then I open your score and I don't have the font, so Neanes falls back to the current EZ Psaltica font. In that case, your custom positioning may not look correct with the other font. Similarly, if you have carefully arranged line breaks and page breaks based on the Stathis font, they would be lost when Neanes falls back to the default font, and breaks may happen in odd places.

1. Are you okay with the font being bundled with Neanes itself (i.e. placed in the `assets/fonts` directory and distributed with the application)?
2. SBMuFL is still an evolving (proposed) standard. If it's bundled with Neanes, I'd like to have a plan for maintenance and updates to this font. Which application was used to create it? Is there a source file that you could provide, such as `sdf` or `ufo`?
3. We should generate a metadata json file specifically for this font, although currently it seems to work well with the Neanes metadata file.

The last thing is the question of whether the original font creator would care that we bundle it. Based on this [Psaltologion post](https://analogion.com/forum/index.php?threads/%CE%A8%CE%B1%CE%BB%CF%84%CE%B9%CE%BA%CE%AE-%CF%83%CE%B7%CE%BC%CE%B1%CE%B4%CE%BF%CF%83%CE%B5%CE%B9%CF%81%CE%AC-stathis-series-%CF%80%CF%81%CE%BF%CF%83%CF%86%CE%AD%CF%81%CE%B5%CF%84%CE%B1%CE%B9.9197/#post-252084), it sounds like the original font creators will not mind us including the font. 

> Ὁ Ὁμότιμος Καθηγητὴς τοῦ Πανεπιστημίου Ἀθηνῶν καὶ Διευθυντὴς τοῦ Ἱδρύματος Βυζαντινῆς Μουσικολογίας κ. Γρηγόριος Στάθης προφρόνως προσφέρει «εἰς κοινὸν ὄφελος» μιὰ ἔκδοση τῆς περίφημης σημαδοσειρᾶς Stathis Series, ποὺ ἔχει σχεδιάσει ὁ ἴδιος καὶ χρησιμοποιεῖται ὣς τώρα ἀποκλειστικὰ στὶς ἐκδόσεις τοῦ Ἱδρύματος Βυζαντινῆς Μουσικολογίας.

There is a request for attribution, but only for official publications. I think we can add a note to the `README.md` letting people know that if they use this font they should include an attribution.

> Μ᾽ αὐτὴν τὴν μορφή, λοιπόν, προσφέρεται στοὺς ψάλτες καὶ ἐκδότες ψαλτικῶν φυλλαδῶν, γιὰ νὰ συμβάλλει στὴν καλαισθησία τῶν γραπτῶν τους ἡ περίφημη χάραξη τῶν σημαδιῶν τῆς Ψαλτικῆς (τῆς Νέας Μεθόδου) ποὺ δημιουργήθηκε ἀπ᾽ τὸν καθηγητὴ Γρηγόριο Στάθη μὲ ἔγνοια νὰ ἀποκρυσταλλωθοῦν «οἱ κινήσεις τῆς πέννας ἢ τῆς φτέρας τόσων μεγάλων μελουργῶν καὶ καλλιτεχνῶν κωδικογράφων τῆς ψαλτικῆς παράδοσης μιᾶς ὑπερχιλιετίας!». Παρακαλοῦνται μόνο ὅσοι τυχὸν τὴν χρησιμοποιήσουν σὲ ἐπίσημες ἐκδόσεις νὰ μνημονεύουν εἴτε στὸν πρόλογο εἴτε στὸν κολοφῶνα ἢ σὲ ἄλλο ἁρμόδιο τόπο τὸν δημιουργὸ καὶ δωροδότη καὶ τὸ συγκεκριμένο ὄνομά της Stathis Series.

**Comment by neobabis** [comment]
Great. Of course, integrating it into Neanes is the way to go. However, that would require some changes to the app, like allowing users to choose their music font, and I wasn't sure if that was something we intended to add to the main app.

1. Absolutely, I have no issue with it being integrated.
2. I used FontForge and based it on the original Neanes font to create Stathis. I don't have any other editing files besides the font itself.

Unfortunately, I'm currently unable to dedicate more time to this.

Overall, the font works very well. The only issue I encountered, which required adjustment compared to the original font, was the combination of 'oligon + kentima with psifiston.'
<img width="48" alt="oligon_kentima" src="https://github.com/neanes/neanes/assets/48261152/ed507ac9-e10a-4f4a-a839-9e4659808ce2">


I didn't encounter the problem you mentioned about falling back to the current EZ Psaltica font.

Note: As mentioned in another thread, it would be useful to add old notations (isaki, strepton, tromikon, ligizma, etc.), as many books still use them and they were readily available in the fonts.
